itests/hive-unit/src/test/java/org/apache/hadoop/hive/ql/parse/TestReplicationScenariosAcidTables.java

```diff
@@ -138,6 +141,211 @@ public void tearDown() throws Throwable {
+    assertTrue(fs.exists(new Path(dumpPath, ReplAck.FAILOVER_READY_MARKER.toString())));
+    assertTrue(fs.exists(new Path(dumpPath, FailoverMetaData.FAILOVER_METADATA)));
+
+    //Since the failover start config is disabled and previous valid dump directory contains _failover_ready marker file
+    //So, this dump iteration will perform bootstrap dump instead of incremental.
+    dumpData = primary.dump(primaryDbName);
+    DumpMetaData dmd = new DumpMetaData(new Path(dumpData.dumpLocation, ReplUtils.REPL_HIVE_BASE_DIR), conf);
+    assertTrue(dmd.getDumpType() == DumpType.BOOTSTRAP);
+  }
+
+  @Test
+  public void testFailoverDuringDumpWithPreviousFailed() throws Throwable {
+    HiveConf primaryConf = primary.getConf();
+    TxnStore txnHandler = TxnUtils.getTxnStore(primary.getConf());
+    WarehouseInstance.Tuple dumpData = null;
+    List<String> failoverConfigs = Arrays.asList("'" + HiveConf.ConfVars.HIVE_REPL_FAILOVER_START + "'='true'");
+    dumpData = primary.run("use " + primaryDbName)
+            .run("create table t1 (id int) clustered by(id) into 3 buckets stored as orc " +
+                    "tblproperties (\"transactional\"=\"true\")")
+            .run("create table t2 (rank int) partitioned by (name string) tblproperties(\"transactional\"=\"true\", " +
+                    "\"transactional_properties\"=\"insert_only\")")
+            .dump(primaryDbName, failoverConfigs);
+
+    //This dump is not failover ready as target db can be used for replication only after first incremental load.
+    FileSystem fs = new Path(dumpData.dumpLocation).getFileSystem(conf);
+    Path dumpPath = new Path(dumpData.dumpLocation, ReplUtils.REPL_HIVE_BASE_DIR);
+    assertFalse(fs.exists(new Path(dumpPath, ReplAck.FAILOVER_READY_MARKER.toString())));
+    assertFalse(MetaStoreUtils.isDbBeingFailedOver(primary.getDatabase(primaryDbName)));
+
+    replica.load(replicatedDbName, primaryDbName)
+            .run("use " + replicatedDbName)
+            .run("show tables")
+            .verifyResults(new String[]{"t1", "t2"})
+            .run("repl status " + replicatedDbName)
+            .verifyResult(dumpData.lastReplicationId);
+
+    primary.run("use " + primaryDbName)
+            .run("insert into t1 values(1)")
+            .run("insert into t2 partition(name='Bob') values(11)")
+            .run("insert into t2 partition(name='Carl') values(10)");
+
+    FailoverMetaData failoverMD = new FailoverMetaData(dumpPath, conf);
+    Long cursorPoint = primary.getCurrentNotificationEventId().getEventId();
+    failoverMD.setCursorPoint(cursorPoint);
+    failoverMD.setOpenTxns(Arrays.asList(1L, 2L, 3L));
+    failoverMD.setAbortedTxns(new ArrayList<>());
+    Long failoverEventId = primary.getCurrentNotificationEventId().getEventId();
+    failoverMD.setEventId(failoverEventId);
+    failoverMD.setTxnsWithoutLocks(new ArrayList<>());
+    failoverMD.write();
+    assertTrue(fs.exists(new Path(dumpPath, FailoverMetaData.FAILOVER_METADATA)));
```

Comment:
What failure case are you trying out here? Could you please explain?

Reply:
This UT is to test the case in which dump fails while dumping the events and after capturing the failover metadata. So, the next dump operations should use the same metadata info in next iteration.